### ISSUE TITLE
LIVENODING 1059 / SV Processing Animated Point Cloud Data

### ISSUE THREAD

**Issue by enzyme69** [enzyme69/blendersushi#447]
If you ever wanted to process a bunch of sequence of CSV data, you can already able to do that rather easily using SV nodes.

I will be using **RADIOHEAD - House of Cards** data, you can download from here:
https://github.com/dataarts/radiohead

The data itself already has a nicely cleaned up data ready to use. This is an example point cloud that you can also observe using Processing app, but we will be using Blender.

**Comment by enzyme69** [comment]
The setup itself is a rather simple procedures:
- Load all the Text CSV data into Blender data blocks
- Read the CSV using SV nodes
- Use a simple Python to simply load the correct CSV, as frame got updated

I recommend you to export everything as Alembic first. 1.6 GB per frame become just over 2 GB of data for 1000 of CSVs.

**Comment by javismiles** [comment]
thank you very much this is great, would it be possible to get the blender file of this example you showed on your vid here? https://www.youtube.com/watch?v=8lSllfH31Kc
,  thank you so much, awesome work! :)

**Comment by javismiles** [comment]
thing is dear Jimmy, you are clearly an expert in Blender (way more than me anyway :) ), and you go so fast in the video, 
also it's not clear what's the name of the AddOn you refer about,
any change to get that AddOn name and location and also the blender file of your example? thank you so so much :D

**Comment by javismiles** [comment]
is sverchok the addOn you refer about? looks like it still doenst work in 2.8 and maybe that's why you did all this in 2.79 then? if sverchok is the addon, is it essential to do all of this?

**Comment by javismiles** [comment]
oh but I now see this
https://www.youtube.com/watch?v=ft-HBy-gFR8
so does this mean that everchok is already ok in 2.8? so can we do all of this in 2.8 now? thank you for your help :)

**Comment by javismiles** [comment]
yes  I successfully installed now sverchok for 2.8 with these instructions
https://github.com/nortikin/sverchok/issues/2326
it's installed well now, 
so this means that I should be able to do your example in 2.8, right?
would it be possible to get the blender file of your example to learn well your node setup? thank you so much :)

**Comment by javismiles** [comment]
so Jimmy,
I managed to make all the nodes following your youtube video, in 2.8, all of them working great,
except when I arrive to the very last one, the node of type  : Exec Node Mod,

I cannot find that one anywhere, how do I create it? thank you so so much

**Comment by javismiles** [comment]
I search all around but cant find anywhere info about how to create an Exec Node Mod node, all the rest I did it and it all works well, hopefully you can help me find out how to create this last node, the Exec Node Mod, thank you so much :)

**Comment by javismiles** [comment]
I found it, it was in Numbers area,
thank you I got it now, and it all works well now,
I implemented exactly what you show in https://www.youtube.com/watch?v=8lSllfH31Kc
great tutorial

my next question now is, at the moment we are replacing each point
with a little sphere,
there must be a way to create a more continuous mesh from all the points,
how can we transform each point cloud of each frame into a more continuous
shape? into a smoother continuous mesh? there must be a way

thank you for any help :)
and congratulations for your great work :)

**Comment by javismiles** [comment]
another question, in latest blender 2.8, after setting up all of this
https://www.youtube.com/watch?v=8lSllfH31Kc
if I render 1 single frame, it all goes well, it renders it perfectly,
but if I use the "Render Animation" blender option, blender crashes and exits

why may this be happening? Im using a powerful Dell xps 15 9550 laptop with 16GB Ram and an Nvidia card with GPU etc

**Comment by javismiles** [comment]
so I can render individual frames, and download them,
but when I try to render them all at once with the render animation option blender always crashes,
could this be an issue with the interaction between b28 and the addon, or just an issue with memory? strange that I can render with no issues
single frames one at a time, but not the animation
thank you so much

**Comment by javismiles** [comment]
it seems to be rendering two identical frames, and then the next two frames change, then again two identical ones, then two different ones, then two identical ones, weird, but when scrubbing through the timeline all the frames are different, all good, but when rendering the animation it renders two identical ones, then two different ones, then 2 identical ones, etc

**Comment by javismiles** [comment]
weird because when I play the animation within blender, it plays perfect, all the frames change as they should, its only when I export to PNGs the animation that weirdly every 2 frames appear identical, every next 2 are different, next 2 are identical, etc and the animation basically is missing half of the changes

**Comment by javismiles** [comment]
very weird, now I tried to export to movie, AVI Jpeg, 10 frames, at 10FPS, 
what I get is an AVI movie with 25 frames, which include the 10 frames that should be exported, they are there, but each is duplicated once or twice, so I get 25 frames that include the 10 correct frames and duplications of them, even though to begin with I didnt request 25 frames, I set 10FPS and a frame range of 1 to 10 also

**Comment by javismiles** [comment]
all right, found a trick, if to export the animation of the 10 frames, I set 20 frames to be exported, then it seems that the 10 frames are then exported correctly followed by 10 static frames after that, really weird but this trick worked although we shouldnt need to do that ;)

apart from that please tell me how I could transform the cloud of little spheres into a more continuous smooth mesh please, thank you :)

**Comment by javismiles** [comment]
so now I have it all working great in blender 2.8, all perfect, the one thing left for me is how to now transform all the dots (spheres now)
into a more continuous mesh rather than a set of separated spheres,
I wrote to the Sverchok team and they told me:
"To make united mesh - Apply matrix node with flag 'join mesh' or apply matrix node + node mesh join."

and I tried to do that, however nothing happens, the spheres of the points still stay separated, so that's my challenge now,
how to unify the spheres/points into a solid mesh, really important for me, any tips I would be super grateful :)

**Comment by enzyme69** [comment]
Hi @javismiles I only found out you posted question in here. So I am trying to do something similar at the moment. Blender 2.80 tends to crash when dealing with Sverchok or Animation Nodes. SO if you can use Blender 2.79 and somewhat transfer that to 2.80....  but I am actually working on similar thing: sequence of point cloud and trying to "mesh" it using SV nodes. 

I'll come back to you...

**Comment by enzyme69** [comment]
@javismiles Yes you need to use Apply Matrix node to join those spheres but use the correct one, there are 2 Apply Matrix node...

<img width="534" alt="Screen Shot 2019-05-26 at 14 14 30" src="https://user-images.githubusercontent.com/3952687/58378587-9adaf500-7fc0-11e9-83bd-3488e27bed3f.png">
